File: Data Structures & Algorithms/sliding-window-maximum/submission-1.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        Deque<Integer> d = new ArrayDeque();
        int[] res = new int[nums.length - k + 1];

        for (int i = 0; i < nums.length; i++) {
            // check if index at the front of the deque is now outside the window <= i-k
            if (!d.isEmpty() && d.peekFirst() <= i - k) {
                d.pollFirst();
            }

            // evict from the back while it is smaller
            while (!d.isEmpty() && nums[d.peekLast()] < nums[i]) {
                d.pollLast();
            }
            d.addLast(i);

            // record the max
            if (i >= k - 1) {
                res[i -k +1] = nums[d.peekFirst()];
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/sort-array-by-parity/submission-1.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        even,odd = [], []

        for n in nums:
            if n & 1:
                odd.append(n)
            else:
                even.append(n)
        
        index = 0
        for e in even:
            nums[index] = e
            index +=1
        for o in odd:
            nums[index] = o
            index +=1
        return nums